### ISSUE TITLE
Parsing error System.ArgumentOutOfRangeException: Length cannot be less than zero.

### ISSUE THREAD

**Issue by SystemsModelling** [rubberduck-vba/Rubberduck#2931]
Is there a way of knowing from this which source line it was at? I could post the code then.

```
2017-03-27 14:04:54.1281;TRACE-2.0.13.32288;Rubberduck.Parsing.VBA.ComponentParseTask;Starting ParseTaskID e9c6ddcc-819c-4db4-8626-4b9c928559a5 on thread 16.;
2017-03-27 14:04:54.1561;ERROR-2.0.13.32288;Rubberduck.Parsing.VBA.ComponentParseTask;Exception thrown in thread 16, ParseTaskID e9c6ddcc-819c-4db4-8626-4b9c928559a5.;System.ArgumentOutOfRangeException: Length cannot be less than zero.
Parameter name: length
   at System.String.Substring(Int32 startIndex, Int32 length)
   at Rubberduck.Parsing.Preprocessing.LivelinessExpression.MarkAsDead(String code) in C:\Users\Mathieu\Documents\GitHub\Rubberduck (main)\Rubberduck\Rubberduck.Parsing\Preprocessing\LivelinessExpression.cs:line 26
   at Rubberduck.Parsing.Preprocessing.LivelinessExpression.Evaluate() in C:\Users\Mathieu\Documents\GitHub\Rubberduck (main)\Rubberduck\Rubberduck.Parsing\Preprocessing\LivelinessExpression.cs:line 21
   at Rubberduck.Parsing.Preprocessing.ConditionalCompilationIfExpression.Evaluate() in C:\Users\Mathieu\Documents\GitHub\Rubberduck (main)\Rubberduck\Rubberduck.Parsing\Preprocessing\ConditionalCompilationIfExpression.cs:line 48
   at Rubberduck.Parsing.Preprocessing.ConditionalCompilationBlockExpression.<>c.<Evaluate>b__2_0(IExpression child) in C:\Users\Mathieu\Documents\GitHub\Rubberduck (main)\Rubberduck\Rubberduck.Parsing\Preprocessing\ConditionalCompilationBlockExpression.cs:line 17
   at System.Linq.Enumerable.WhereSelectListIterator`2.MoveNext()
   at System.String.Join(String separator, IEnumerable`1 values)
   at Rubberduck.Parsing.Preprocessing.ConditionalCompilationBlockExpression.Evaluate() in C:\Users\Mathieu\Documents\GitHub\Rubberduck (main)\Rubberduck\Rubberduck.Parsing\Preprocessing\ConditionalCompilationBlockExpression.cs:line 17
   at Rubberduck.Parsing.Preprocessing.VBAPreprocessor.Execute(String moduleName, String unprocessedCode, CancellationToken token) in C:\Users\Mathieu\Documents\GitHub\Rubberduck (main)\Rubberduck\Rubberduck.Parsing\Preprocessing\VBAPreprocessor.cs:line 24
   at Rubberduck.Parsing.VBA.ComponentParseTask.RewriteAndPreprocess(CancellationToken token) in C:\Users\Mathieu\Documents\GitHub\Rubberduck (main)\Rubberduck\Rubberduck.Parsing\VBA\ComponentParseTask.cs:line 184
   at Rubberduck.Parsing.VBA.ComponentParseTask.Start(CancellationToken token) in C:\Users\Mathieu\Documents\GitHub\Rubberduck (main)\Rubberduck\Rubberduck.Parsing\VBA\ComponentParseTask.cs:line 51
```

**Comment by Vogel612** [comment]
Has been mentioned already in #2918 and #2912 ,,,

**Comment by SystemsModelling** [comment]
Sorry, I missed that. Do you want to delete this? Or leave it in as the error is specifically shown in the subject?

**Comment by MDoerner** [comment]
This has long been fixed (since PR #3001).

There is no string manipulation anymore in the class that caused the exception.
